src/python/pants/backend/python/goals/pytest_runner.py

```diff
@@ -311,14 +313,32 @@ async def setup_pytest_for_target(
             f"--cov-config={coverage_config.path}",
             *cov_args,
         ]
+        pants_add_opts.extend(["--cov-report", "--cov-config"])  # --cov not there as multi-allowed.
+
+    # Get extra PYTEST_ADDOPTS env var from each possible sources.
+    extra_add_opts = []
+    if "PYTEST_ADDOPTS" in field_set_extra_env:
+        extra_add_opts.append(field_set_extra_env["PYTEST_ADDOPTS"])
+    if "PYTEST_ADDOPTS" in test_extra_env.env:
+        extra_add_opts.append(test_extra_env.env["PYTEST_ADDOPTS"])
+    add_opts.extend(extra_add_opts)
+
+    # Warn the user if there is an option in PYTEST_ADDOPTS already set by Pants.
+    for popt in pants_add_opts:
+        if any(popt in eopt for eopt in extra_add_opts):
+            logger.warning(
+                f"The option '{popt}' is set by Pants but also in PYTEST_ADDOPTS environment variable. This can cause unexpected behavior."
```

Comment:
So, I played around with pytest 7.0.1 which is what main currently ships with by default:
https://github.com/erjac77/pants/blob/40e17902092a9b7414473f99bf32e3f3c84e5921/src/python/pants/backend/python/subsystems/pytest.py#L62-L74

It has much more variety in what it accepts. FWICT, there are these variants:
1. `--junit-xml` or `--junitxml`
2. `=` or ` `
3. `-o` or `--override-ini`

For the last, you currently ignore the `-o` prefix to `junit_family` and that's probably fine; so probably this can be ignored. Likewise for 2 you can ignore the introduction of the value separator and you do; so things are still good. For the 1st though, that would need to be checked too.